modules/core/src/main/java/io/fluo/recipes/transaction/RecordingTransactionBase.java

```diff
@@ -41,46 +53,82 @@ public void setWeakNotification(Bytes row, Column col) {
 
   @Override
   public void set(Bytes row, Column col, Bytes value) throws AlreadySetException {
-    txLog.addSet(row, col, value);
+    txLog.filteredAdd(LogEntry.newSet(row, col, value), filter);
     txb.set(row, col, value);
   }
 
   @Override
   public void delete(Bytes row, Column col) {
-    txLog.addDelete(row, col);
+    txLog.filteredAdd(LogEntry.newDelete(row, col), filter);
     txb.delete(row, col);
   }
 
   @Override
   public Bytes get(Bytes row, Column col) {
-    return txb.get(row, col);
+    Bytes val = txb.get(row, col);
+    if (val != null) {
+      txLog.filteredAdd(LogEntry.newGet(row, col, val), filter);
+    }
+    return val;
   }
 
   @Override
   public Map<Column, Bytes> get(Bytes row, Set<Column> columns) {
-    return txb.get(row, columns);
+    Map<Column, Bytes> colVal = txb.get(row, columns);
+    for (Map.Entry<Column, Bytes> entry : colVal.entrySet()) {
+      txLog.filteredAdd(LogEntry.newGet(row, entry.getKey(), entry.getValue()), filter);
+    }
+    return colVal;
   }
 
   @Override
   public Map<Bytes, Map<Column, Bytes>> get(Collection<Bytes> rows, Set<Column> columns) {
-    return txb.get(rows, columns);
+    Map<Bytes, Map<Column, Bytes>> rowColVal = txb.get(rows, columns);
+    for (Map.Entry<Bytes, Map<Column, Bytes>> rowEntry : rowColVal.entrySet()) {
+      for (Map.Entry<Column, Bytes> colEntry : rowEntry.getValue().entrySet()) {
+        txLog.filteredAdd(
+            LogEntry.newGet(rowEntry.getKey(), colEntry.getKey(), colEntry.getValue()), filter);
+      }
+    }
+    return rowColVal;
   }
 
   @Override
   public RowIterator get(ScannerConfiguration config) {
-    return txb.get(config);
+    RowIterator rowIter = txb.get(config);
+    while ((rowIter != null) && rowIter.hasNext()) {
```

Comment:
Seems like this will consume all data and give the user back an iterator where hasNext() returns false.  I didn't see a test.

This could wrap the iterator and only add what the users consumes.... whatever is not consumed would not be added to the txlog